lib/index.js

```diff
@@ -97,69 +77,32 @@ function clearRuntimeIndex() {
 }
 
 function rebuildPagesIndex() {
-  return new Promise((resolve, reject) => {
-    clearPagesIndex()
-      .then(() => {
-        return getShortIndex();
-      })
-      .then(() => {
-        resolve();
-      })
-      .catch((err) => {
-        reject(err);
-      });
-  });
+  return clearPagesIndex().then(() => getShortIndex());
 }
 
 // If the variable is not set, read the file and set it.
 // Else, just return the variable.
 function getShortIndex() {
-  return new Promise((resolve, reject) => {
-    if (shortIndex) {
-      resolve(shortIndex);
-    } else {
-      readShortPagesIndex()
-        .then((idx) => {
-          resolve(idx);
-        })
-        .catch((err) => {
-          reject(err);
-        });
-    }
-  });
+  if (shortIndex) {
+    return Promise.resolve(shortIndex);
+  }
+  return readShortPagesIndex();
 }
 
 // Read the index file, and load it into memory.
 // If the file does not exist, create the data structure, write the file,
 // and load it into memory.
 function readShortPagesIndex() {
-  return new Promise((resolve, reject) => {
-    fs.readFile(shortIndexFile, 'utf8')
-      .then((idx) => {
-        // File is present, so just parse
-        // and set the shortIndex variable,
-        // then resolve the value.
-        idx = JSON.parse(idx);
-        shortIndex = idx;
-        return resolve(idx);
-      })
-      .catch(() => {
-        // File is not present; we need to create the index.
-        let idx = buildShortPagesIndex();
-        if (Object.keys(idx).length > 0) {
-          fs.writeFile(shortIndexFile, JSON.stringify(idx))
-            .then(() => {
-              shortIndex = idx;
-              return resolve(idx);
-            })
-            .catch((err) => {
-              reject(err);
-            });
-        } else {
-          return resolve(idx);
-        }
-      });
-  });
+  return fs.readJson(shortIndexFile)
+    .catch(() => {
```

Comment:
Please move the catch block after `then()`.

Reply:
Actually can't do that here because this `catch()` resolves with created index that gets consumed by next `then()`; think of it as async else block.